File: qqbot-client/src/main/java/com/mieai/qqbot/client/QqArkMessageRequest.java
package com.mieai.qqbot.client;

import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;

public record QqArkMessageRequest(QqMessageTargetType targetType, String targetId, int templateId,
        List<Map<String, String>> values, Optional<String> replyMessageId, Optional<String> replyEventId,
        int messageSequence) {
    public QqArkMessageRequest {
        Objects.requireNonNull(targetType, "targetType must not be null");
        if (targetId == null || targetId.isBlank()) throw new IllegalArgumentException("targetId is invalid");
        if (templateId < 1) throw new IllegalArgumentException("templateId must be positive");
        values = List.copyOf(Objects.requireNonNull(values, "values must not be null")).stream().map(Map::copyOf).toList();
        Objects.requireNonNull(replyMessageId, "replyMessageId must not be null");
        Objects.requireNonNull(replyEventId, "replyEventId must not be null");
        if (messageSequence < 1) throw new IllegalArgumentException("messageSequence must be positive");
    }
}

File: qqbot-client/src/main/kotlin/com/mieai/qqbot/client/QqArkMessageRequest.kt
package com.mieai.qqbot.client

class QqArkMessageRequest(
    val targetType: QqMessageTargetType,
    val targetId: String,
    val templateId: Int,
    values: List<Map<String, String>>,
    val replyMessageId: String?,
    val replyEventId: String?,
    val messageSequence: Int,
) {
    val values: List<Map<String, String>> = values.map { it.toMap() }

    init {
        require(targetId.isNotBlank()) { "targetId is invalid" }
        require(templateId >= 1) { "templateId must be positive" }
        require(messageSequence >= 1) { "messageSequence must be positive" }
    }
}
